### PR TITLE
mariadb: fix mytop script

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
```
~/nixpkgs> nix-shell -p mysql.client --run mytop
/tmp/nix-shell-632-0/rc: /nix/store/fzfx1sr6pfrqhqy7n7jh89sbxr4v7da5-mariadb-client-10.2.17-bin/bin/mytop: /usr/bin/perl: bad interpreter: No such file or directory
```

@volth Is this an acceptable way to fix this? Do I need `propagatedBuildInputs` here?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by volth** [comment]
lgtm

**Comment by aanderse** [comment]
@GrahamcOfBorg build perlPackages.DBDmysql

**Comment by aanderse** [comment]
Any darwin folks who could take a look at `perlPackages.DBDmysql` possibly? @matthewbauer Am I mistaken in thinking you're a darwin person?

**Comment by aanderse** [comment]
@lilyball As somone who has expressed interest on making things better on the darwin side (https://discourse.nixos.org/t/suggestion-github-label-for-prs-for-darwin-packages/3458) is there any chance I could get you to take a look at the failure here? :heart:

**Comment by lilyball** [comment]
I don't know exactly what's going on, but the error is coming from `mysql.connector-c`. Specifically, the `mysql_config` tool it vends has two issues:

1. The perl package tries to pass the `--embedded` flag to it but it doesn't support this flag, and
2. The `mysql_config --libs` output has that stray `/nix/store/9mf51m54d2hmj78h43n9pi31wnzdn3v5-libiconv-osx-10.12.6/lib/libiconv.dylib` argument with no flag prefix.

For contrast, the `mysql_config` vended from the root `mysql` package supports `--embedded` and does not have that stray argument.

The first error may be ignorable, but the second is obviously a problem, and I don't know what's going on there. `libiconv` is one of the `buildInputs` to the `mysql.connector-c` derivation, but aside from its presence there, I don't know why it's showing up in the `--libs` output in this fashion. Maybe the maintainer @globin might have an idea.

**Comment by aanderse** [comment]
@lilyball Thanks for taking a look, I really appreciate it!

I have no access to a mac so I can't do anything about this. I really hate to make this a `platforms.linux` only fix, but I just don't know how else to proceed... As we get closer to 19.09 freeze the number of available options decreases.

**Comment by lilyball** [comment]
I do not think "Cylance incorrectly flags this as a problem" is justification for "make this a linux-only fix". Anyone affected by this should be complaining to Cylance about their false positives. Antivirus false positives can happen on any platform.

**Comment by aanderse** [comment]
@lilyball Well this error isn't related to the Cylance problem. As far as anyone else reporting the issue I assume either `mytop` isn't used but anyone using nixpkgs, or this problem is/was a regression on all platforms. Given the nature of the fix I assume the former.

My fix works on `platforms.linux` but as mentioned I would rather not make the fix conditional on `platforms`.

**Comment by lilyball** [comment]
Oh I'm terribly sorry, I thought I was in a different tab. I retract my comment entirely.

**Comment by aanderse** [comment]
@lilyball No problem :smile:

File: pkgs/servers/sql/mariadb/default.nix [hunk 1]
```diff
@@ -181,6 +182,17 @@ everything = stdenv.mkDerivation (common // {
     sed -i 's/-mariadb/-mysql/' "$out"/bin/galera_new_cluster
   '';
 
+  postFixup =
+    let
+      mytopEnv = perl.withPackages (p: with p; [ DataDumper DBDmysql DBI TermReadKey ]);
+    in
+      # perlPackages.DBDmysql is broken on darwin
+      optionalString (!stdenv.isDarwin) ''
+        sed -i 's|/usr/bin/perl|${mytopEnv.interpreter}|' $out/bin/mytop
```
Comment (Mic92) [inline pkgs/servers/sql/mariadb/default.nix]: Use `patchShebangs` here by adding mytopEnv to `buildInputs`.

Comment (Mic92) [reply pkgs/servers/sql/mariadb/default.nix]: Also using `wrapProgramInPlace` from `makeWrapper` to extend `PATH` is the better option to add `less` and `more` over sed. Looking at the code only either less or more is required. I would go for less.

Comment (aanderse) [reply pkgs/servers/sql/mariadb/default.nix]: Thank you very much @Mic92! How does this newest revision look?

**Pull request merged**